extensions/data-transfer/portability-data-transfer-backblaze/src/test/java/org/datatransferproject/datatransfer/backblaze/photos/BackblazePhotosImporterTest.java

```diff
@@ -42,119 +43,120 @@
+  IdempotentImportExecutor executor;
+  TokenSecretAuthData authData;
+  BackblazeDataTransferClient client;
+
+  @BeforeEach
+  public void setUp() {
+    monitor = mock(Monitor.class);
+    dataStore = mock(TemporaryPerJobDataStore.class);
+    streamProvider = mock(ImageStreamProvider.class);
+    clientFactory = mock(BackblazeDataTransferClientFactory.class);
+    executor = mock(IdempotentImportExecutor.class);
+    authData = mock(TokenSecretAuthData.class);
+    client = mock(BackblazeDataTransferClient.class);
+  }
+
+  @TempDir
+  public Path folder;
+
+  @Test
+  public void testNullData() throws Exception {
+    BackblazePhotosImporter sut =
+        new BackblazePhotosImporter(monitor, dataStore, streamProvider, clientFactory);
+    ImportResult result = sut.importItem(UUID.randomUUID(), executor, authData, null);
+    assertEquals(ImportResult.OK, result);
+  }
+
+  @Test
+  public void testNullPhotosAndAlbums() throws Exception {
+    PhotosContainerResource data = mock(PhotosContainerResource.class);
+    when(data.getAlbums()).thenReturn(null);
+    when(data.getPhotos()).thenReturn(null);
+
+    BackblazePhotosImporter sut =
+        new BackblazePhotosImporter(monitor, dataStore, streamProvider, clientFactory);
+    ImportResult result = sut.importItem(UUID.randomUUID(), executor, authData, data);
+    assertEquals(ImportResult.OK, result);
+  }
+
+  @Test
+  public void testEmptyPhotosAndAlbums() throws Exception {
+    PhotosContainerResource data = mock(PhotosContainerResource.class);
+    when(data.getAlbums()).thenReturn(new ArrayList<>());
+    when(data.getPhotos()).thenReturn(new ArrayList<>());
+
+    BackblazePhotosImporter sut =
+        new BackblazePhotosImporter(monitor, dataStore, streamProvider, clientFactory);
+    ImportResult result = sut.importItem(UUID.randomUUID(), executor, authData, data);
+    assertEquals(ImportResult.OK, result);
+  }
+
+  @Test
+  public void testImportPhoto() throws Exception {
+    String dataId = "dataId";
+    String title = "title";
+    String photoUrl = "photoUrl";
+    String albumName = "albumName";
+    String albumId = "albumId";
+    String response = "response";
+    UUID jobId = UUID.randomUUID();
+    PhotoModel photoModel = new PhotoModel(title, photoUrl, "", "", dataId, albumId, false);
+    PhotosContainerResource data = new PhotosContainerResource(Collections.emptyList(),
+        Collections.singletonList(photoModel));
+
+    when(executor.getCachedValue(albumId)).thenReturn(albumName);
+
+    HttpURLConnection connection = mock(HttpURLConnection.class);
+    when(connection.getInputStream()).thenReturn(IOUtils.toInputStream("photo content", "UTF-8"));
+    when(streamProvider.getConnection(photoUrl)).thenReturn(connection);
+
+    when(client.uploadFile(eq("Photo Transfer/albumName/dataId.jpg"), any())).thenReturn(response);
+    when(clientFactory.getOrCreateB2Client(jobId, authData)).thenReturn(client);
+
+    File file = folder.toFile();
```

Comment:
Why are we changing `newFile` to `toFile`?

Reply:
The previous implementation was @Rule based in JUnit4 , the .newFile() previously as we see is invoked on TemporaryFolder object which is not supported anymore in JUnit5, instead its converted to a path which is annotated with @TempDir (Junit5) which supplies a temp dir location (cleanups are maintained by JUnit5), .toFile() method returns file object representing the file object representing the default path of the temp dir defined.

Reply:
Oh I see! I totally missed the fact we also changed the field type from `TemporaryFolder` to `Path`. Then I got confused on converting a _directory_ path to a file (as opposed to creating a _file_ in a directory). Looks like in our case both approaches work because the only time we use it in the test [we take its size and pass the file to a mock](https://github.com/google/data-transfer-project/blob/9cc43140a6fea2190dd66cae4235700e8a622f33/extensions/data-transfer/portability-data-transfer-backblaze/src/main/java/org/datatransferproject/datatransfer/backblaze/photos/BackblazePhotosImporter.java#L110-L115).